docs/architecture.md

```diff
@@ -37,21 +37,26 @@ least-recently-used eviction policy.
 
 #### Get
 
-Retrieve a contract's value
+Retrieve a contract's value. The contract webassembly is hashed, and the hash is 
+[converted](https://github.com/sanity/locutus/blob/master/src/main/kotlin/locutus/tools/math/Location.kt#L23) 
+to a location on the ring. The Get request is then "greedy routed" towards that
+location. If/when the data is found it is returned along the same path to
+the requestor, potentially being cached by peers along the way.

```

Comment:
I have a question regarding this operation, is more of a matter of clarification.  Once a node joins the network (adding some known initial nodes/peers) and wants to fetch certain data (in our case the wasm contracts) from an other node, it has to know in advance the location of the target (and by extension the hash of the contract) in order for the greedy algorithm to be able to reach that location.

So in the case of open applications where you want people to join you would publish the key so other peers can fetch the contract and connect to your application. Is this correct?

Reply:
> Once a node joins the network (adding some known initial nodes/peers) and wants to fetch certain data (in our case the wasm contracts) from an other node, it has to know in advance the location of the target (and by extension the hash of the contract) in order for the greedy algorithm to be able to reach that location.

Yes - at a minimum the requestor must know the hash of the contract - which can be mapped to a location. Of course if you have the contract you can generate the hash.

> So in the case of open applications where you want people to join you would publish the key so other peers can fetch the contract and connect to your application. Is this correct?

Hmm, that depends on what you mean by "application". I think of an "application" as a use case, so microblogging is an application, group chat would be another application, an online marketplace would be another application. Applications are built on top of the key/value store.

But I think the answer is "yes", people will need some way to initially "discover" contracts. These might be defaults provided with the application (similar to how Classic Freenet has default "freesites"), or in an IM-type application people might discover their friend's contract addresses through some other means.

Reply:
Ok that answers my question, thanks.